src/Hl7.Fhir.Base/Specification/Snapshot/SnapshotGeneratorExtensions.cs

```diff
@@ -88,6 +88,20 @@ public static void RemoveAllConstrainedByDiffExtensions<T>(this IEnumerable<T> e
             }
         }
 
+        internal static bool HasAnyNonInheritableExtensions(this Element element)
+        {
+            if (element == null) return false;
+            if (element.HasNonInheritableExtensions()) return true;
+#pragma warning disable CS0618 // Type or member is obsolete
+            return element.Children().OfType<Element>().Any(child => child.HasAnyNonInheritableExtensions());
+#pragma warning restore CS0618 // Type or member is obsolete
+        }
+
+        internal static bool HasNonInheritableExtensions(this IExtendable element)
+        {
+            return element != null && _nonInheritableExtensions.Any(ext => element.GetExtension(ext) != null);
```

Comment:
I am a bit worried about the number of iterations here:  for each element in the profile we go over all `_nonInheritableExtensions`, and for each of those we scan the extension list to match a string. That sounds like N*M*P, where P is a low number.

If we'd make _nonInheritableExtensions a hashset, then we could reverse this bit and do `element.Extensions.Any(e => _nonInheritableExtensions.Contains(e.Url.Value))` (or something along those lines). That would probably save a lot of scans.

Reply:
You were right. Optmizing it as you suggested will increase performance. A quick test in Forge showed a decrease of the load time by 5 - 10%.